media-sound/beets/beets-1.4.9-r3.ebuild

```diff
@@ -0,0 +1,178 @@
+		dev-python/pyyaml[${PYTHON_MULTI_USEDEP}]
+		dev-python/requests[${PYTHON_MULTI_USEDEP}]
+		dev-python/requests-oauthlib[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/six-1.9[${PYTHON_MULTI_USEDEP}]
+		dev-python/unidecode[${PYTHON_MULTI_USEDEP}]
+		>=media-libs/mutagen-1.33[${PYTHON_MULTI_USEDEP}]
+		badfiles? (
+			media-libs/flac
+			media-sound/mp3val
+		)
+		chromaprint? (
+			dev-python/audioread[${PYTHON_MULTI_USEDEP}]
+			dev-python/pyacoustid[${PYTHON_MULTI_USEDEP}]
+			media-libs/chromaprint[tools]
+		)
+		discogs? ( dev-python/discogs-client[${PYTHON_MULTI_USEDEP}] )
+		ffmpeg? ( media-video/ffmpeg:0[encode] )
+		gstreamer? (
+			media-libs/gst-plugins-good:1.0
+			media-libs/gst-plugins-bad:1.0
+		)
+		icu? ( dev-db/sqlite[icu] )
+		lastfm? ( dev-python/pylast[${PYTHON_MULTI_USEDEP}] )
+		mpd? (
+			dev-python/bluelet[${PYTHON_MULTI_USEDEP}]
+			dev-python/python-mpd[${PYTHON_MULTI_USEDEP}]
+		)
+		replaygain? (
+			gstreamer? (
+				dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]
+				media-plugins/gst-plugins-libav:1.0
+			)
+			!gstreamer? ( media-sound/mp3gain )
+		)
+		thumbnail? (
+			dev-python/pyxdg[${PYTHON_MULTI_USEDEP}]
+			|| (
+				dev-python/pillow[${PYTHON_MULTI_USEDEP}]
+				media-gfx/imagemagick
+			)
+		)
+		webserver? (
+			dev-python/flask[${PYTHON_MULTI_USEDEP}]
+			cors? (
+				dev-python/flask-cors[${PYTHON_MULTI_USEDEP}]
+			)
+		)
+	')"
+BDEPEND="dev-python/sphinx"
+
+DOCS=( README.rst docs/changelog.rst )
+
+distutils_enable_tests pytest
+
+python_prepare_all() {
+	distutils-r1_python_prepare_all
+
+	rm_use_plugins() {
+		[[ -n "${1}" ]] || die "rm_use_plugins: No use option given"
+		local use=${1}
+		local plugins=${use}
+		use ${use} && return
+		einfo "no ${use}:"
+		[[ $# -gt 1 ]] && plugins="${@:2}"
+		for arg in ${plugins[@]}; do
+			einfo "  removing ${arg}"
+			if [[ -e "beetsplug/${arg}.py" ]]; then
+				rm beetsplug/${arg}.py || die "Unable to remove ${arg} plugin"
+			fi
+			if [[ -d "beetsplug/${arg}" ]]; then
+				rm -r beetsplug/${arg} || die "Unable to remove ${arg} plugin"
+			fi
+			sed -e "s:'beetsplug.${arg}',::" -i setup.py || \
+				die "Unable to disable ${arg} plugin "
+		done
+	}
+
+	rm_use_plugins chromaprint chroma
+	rm_use_plugins ffmpeg convert
+	rm_use_plugins icu loadext
+	rm_use_plugins lastfm lastgenre lastimport
+	rm_use_plugins mpd bpd mpdstats
+	rm_use_plugins webserver web
+	rm_use_plugins thumbnail thumbnails
+
+	# remove plugins that do not have appropriate dependencies installed
+	for flag in badfiles discogs replaygain; do
+		rm_use_plugins ${flag}
+	done
+
+	if ! use mpd; then
+		rm test/test_player.py || die
+		rm test/test_mpdstats.py || die
+	fi
+	if ! use webserver; then
+		rm test/test_web.py || die "Failed to remove test_web.py"
+	fi
+	if ! use replaygain; then
+		rm test/test_replaygain.py || die
+	fi
+	if ! use ffmpeg; then
+		rm test/test_convert.py || die
+	fi
+	if ! use thumbnail; then
+		rm test/test_thumbnails.py || die
+	fi
```

Comment:
Your message summary "cleanup" doesn't say anything you're doing to this ebuild. Can you print a diff between -r2 and -r3? 

Besides I hate this kind of conditional removing, shouldn't this all just be unconditionally enabled where Gentoo supports it?

Reply:
> Your message summary "cleanup" doesn't say anything you're doing to this ebuild.

Yes wasn't sure how to explain it here the diff.

```diff
--- beets-1.4.9-r2.ebuild       2020-07-06 18:55:05.776887543 +0200
+++ beets-1.4.9-r3.ebuild       2020-07-06 18:55:06.766887901 +0200
@@ -22,11 +22,11 @@ else
 fi
 
 DESCRIPTION="Media library management system for obsessive-compulsive music geeks"
-HOMEPAGE="http://beets.io/ https://pypi.org/project/beets/"
+HOMEPAGE="https://beets.io/ https://pypi.org/project/beets/"
 
 LICENSE="MIT"
 SLOT="0"
-IUSE="badfiles chromaprint discogs doc ffmpeg gstreamer icu lastfm mpd replaygain test thumbnail webserver"
+IUSE="badfiles chromaprint discogs doc ffmpeg gstreamer icu lastfm mpd replaygain test thumbnail webserver cors"
 
 RDEPEND="${DEPEND}"
 DEPEND="
@@ -54,6 +54,7 @@ DEPEND="
                        media-sound/mp3val
                )
                chromaprint? (
+                       dev-python/audioread[${PYTHON_MULTI_USEDEP}]
                        dev-python/pyacoustid[${PYTHON_MULTI_USEDEP}]
                        media-libs/chromaprint[tools]
                )
@@ -85,7 +86,9 @@ DEPEND="
                )
                webserver? (
                        dev-python/flask[${PYTHON_MULTI_USEDEP}]
-                       dev-python/flask-cors[${PYTHON_MULTI_USEDEP}]
+                       cors? (
+                               dev-python/flask-cors[${PYTHON_MULTI_USEDEP}]
+                       )
                )
        ')"
 BDEPEND="dev-python/sphinx"
@@ -131,27 +134,30 @@ python_prepare_all() {
        done
 
        if ! use mpd; then
-               rm test/test_player.py || die
-               rm test/test_mpdstats.py || die
+               rm test/test_player.py || die "Failed to remove test_player.py"
+               rm test/test_mpdstats.py || die "Failed to remove test_mpdstats.py"
        fi
        if ! use webserver; then
                rm test/test_web.py || die "Failed to remove test_web.py"
        fi
+       if ! use replaygain; then
+               rm test/test_replaygain.py || die "Failed to remove test_replaygain.py"
+       fi
+       if ! use ffmpeg; then
+               rm test/test_convert.py || die "Failed to remove test_convert.py"
+       fi
+       if ! use thumbnail; then
+               rm test/test_thumbnails.py || die "Failed to remove test_thumbnails.py"
+       fi
        if use test; then
                # Those test need network
-               rm test/test_art.py || die
-               rm test/test_discogs.py || die
-               rm test/test_embyupdate.py || die
-               rm test/test_lastgenre.py || die
-               rm test/test_spotify.py || die
-               # rm test/test_plexupdate.py
-               rm test/test_thumbnails.py || die
-               # Not working
-               rm test/test_replaygain.py || die
-               # Not working
-               rm test/test_convert.py || die
+               rm test/test_art.py || die "Failed to remove test_art.py"
+               rm test/test_discogs.py || die "Failed to remove test_discogs.py"
+               rm test/test_embyupdate.py || die "Failed to remove test_embyupdate.py"
+               rm test/test_lastgenre.py || die "Failed to remove test_lastgenre.py"
+               rm test/test_spotify.py || die "Failed to remove test_spotify.py"
                # Not working and dropped in master
-               rm test/test_mediafile.py || die
+               rm test/test_mediafile.py || die "Failed to remove test_mediafile.py"
        fi
 }
```

> Besides I hate this kind of conditional removing, shouldn't this all just be unconditionally enabled where Gentoo supports it?

Maybe I don't get it but here the source is enabled by default right ?
So I don't see how we can enable if supported, but please show one example.

Reply:
If it can not be disabled in a sane way provided by upstream, you'll have to enable them unconditionally as upstream wishes.
https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301